File: Day3.js
// Problem 3: Array Manipulator (Arrays, Loops, Array Methods)
// Goal: Practice common array operations.
// Tasks:
// Given an array of numbers: const numbers = [10, 25, 5, 40, 15, 30];
// Find Max: Write a function findMax(arr) that returns the largest number in the array.
// Filter Even: Write a function filterEven(arr) that returns a new array containing only the even numbers.
// Sum All: Write a function sumAll(arr) that returns the sum of all numbers in the array.
// Double Each: Write a function doubleEach(arr) that returns a new array where each number is doubled.
// Find Index: Write a function findIndex(arr, value) that returns the index of the value in the arr, or -1 if not found.
// Concepts Covered: Arrays, Array Methods (forEach, filter, reduce, map, indexOf or find), Loops (for), Functions.

const numbers = [10, 25, 5, 40, 15, 30];

function findMax(arr) {
  return Math.max(...arr);
}


function filterEven(arr) {
  return arr.filter(num => num % 2 === 0);
}


function sumAll(arr) {
  return arr.reduce((sum, num) => sum + num, 0);
}


function doubleEach(arr) {
  return arr.map(num => num * 2);
}


function findIndex(arr, value) {
  return arr.indexOf(value);
}
console.log(findMax(numbers));        
console.log(filterEven(numbers));    
console.log(sumAll(numbers));        
console.log(doubleEach(numbers));    
console.log(findIndex(numbers, 15)); 
console.log(findIndex(numbers, 99)); 

File: Java/Day3.java
// Problem 3: Array Operations (Arrays, Loops)
// Goal: Work with one-dimensional arrays and loops.
// Tasks:
// Create a class ArrayOperations.
// In the main method, declare and initialize an integer array, e.g., int[] numbers = {10, 25, 5, 40, 15, 30};.
// Find Max: Write a loop to find and print the largest number in the array.
// Sum All: Write a loop to calculate and print the sum of all numbers in the array.
// Print Even: Write a loop to print only the even numbers from the array.
// Reverse Array: Create a new array and fill it with the elements of the numbers array in reverse order. Print the reversed array.
// Concepts Covered: Array declaration and initialization, array indexing, for loop, for-each loop (enhanced for loop).


public class ArrayOperations {
    public static void main(String[] args) {
        int[] numbers = {10, 25, 5, 40, 15, 30};

        // 🔹 Find Max
        int max = numbers[0];
        for (int num : numbers) {
            if (num > max) {
                max = num;
            }
        }
        System.out.println("Maximum number: " + max);

        // 🔹 Sum All
        int sum = 0;
        for (int num : numbers) {
            sum += num;
        }
        System.out.println("Sum of all numbers: " + sum);

        // 🔹 Print Even
        System.out.print("Even numbers: ");
        for (int num : numbers) {
            if (num % 2 == 0) {
                System.out.print(num + " ");
            }
        }
        System.out.println();

        // 🔹 Reverse Array
        int[] reversed = new int[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            reversed[i] = numbers[numbers.length - 1 - i];
        }

        System.out.print("Reversed array: ");
        for (int num : reversed) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
